grant-project/zcash-grant-system
# Request for Proposal (Pt. 1 - Models & Admin)

A piece of #84.

### What This Does

This is the first of probably 3 prs for RFPs. This first one adds an admin page for managing RFPs. You can list all, view, edit, create, and delete RFPs. This admin borrows heavily from @AMStrix's work on proposal admin, so thanks for that!

Some things that weren't discussed but I implemented:

* RFPs have categories and briefs, just like proposals. This was kind of a just-in-case measure, since  figured they'd display in very similar ways to proposals.
* In addition to the DRAFT and LIVE statuses, there's a CLOSED status. This will still be publicly viewable so that users can view past proposals, but won't allow new proposal submissions.
* I refactored a lot of code surrounding statuses along the way to be more enum-like.

The next two steps are to add a view to the public site, and then to allow people to create proposals against public RFPs.

### Steps to Test

* Run a db upgrade to get the new tables
* Go to the admin, confirm you see no RFPs in the list
* Create an RFP, confirm the form validates if you have any invalid fields
* Save the new RFP, confirm you see it in the list now as a draft
* Edit the RFP, confirm you now see a "Status" field you can change
* Change some fields and save, confirm they update
* Delete an RFP, confirm it's deleted

## Screenshots

### List

![screen shot 2019-01-29 at 1 08 21 pm](https://user-images.githubusercontent.com/649992/51930128-db868100-23c7-11e9-8ed4-04db15af5046.png)

### Edit

![screen shot 2019-01-29 at 1 08 41 pm](https://user-images.githubusercontent.com/649992/51930165-e80ad980-23c7-11e9-9949-de30e0253d46.png)


### View

![screen shot 2019-01-29 at 1 09 02 pm](https://user-images.githubusercontent.com/649992/51930151-e5a87f80-23c7-11e9-9815-9bdc3b1f8924.png)
